### PR TITLE
Remove diffusers dependency in onnxruntime code

### PR DESCRIPTION
**Pull request by fxmarty**
# What does this PR do?

As per title.

Fixes https://github.com/huggingface/optimum/issues/615

### CODE REVIEW

File: optimum/exporters/onnx/convert.py [hunk 1]
```diff
@@ -51,11 +51,15 @@ def check_dummy_inputs_are_allowed(
...
-    )
+
+    classinfo = (PreTrainedModel,)
+    if is_diffusers_available():
+        classinfo += (ModelMixin,)
+
+    if "diffusers" in str(model.__class__) and not is_diffusers_available():
+        raise ImportError("The pip package `diffusers` is required to export stable diffusion models to ONNX.")
+
+    forward = model.forward if is_torch_available() and issubclass(type(model), classinfo) else model.call
```
Comment (fxmarty) [inline optimum/exporters/onnx/convert.py:62]: this seems very verbose and redundant, not sure if the check `issubclass` is necessary? @michaelbenayoun @echarlaix

Comment (michaelbenayoun) [reply optimum/exporters/onnx/convert.py:62]: Yes, let's simplify this as you suggest!

File: optimum/exporters/onnx/convert.py [hunk 2]
```diff
@@ -154,17 +158,22 @@ def validate_model_outputs(
...
 
-    framework = (
-        "pt" if is_torch_available() and issubclass(type(reference_model), (PreTrainedModel, ModelMixin)) else "tf"
-    )
+    classinfo = (PreTrainedModel,)
+    if is_diffusers_available():
+        classinfo += (ModelMixin,)
+
+    if "diffusers" in str(reference_model.__class__) and not is_diffusers_available():
+        raise ImportError("The pip package `diffusers` is required to validate stable diffusion ONNX models.")
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/convert.py:166]: Maybe make it a function if this is going to be repeated a few times in the functions of this file.

File: optimum/onnxruntime/modeling_decoder.py [hunk 3]
```diff
@@ -623,7 +623,7 @@ def _from_transformers(
         task: Optional[str] = None,
     ) -> "ORTModelDecoder":
         if task is None:
-            task = cls._AUTOMODELS_TO_TASKS[cls.auto_model_class]
+            task = cls._AUTOMODELS_TO_TASKS[cls.auto_model_class.__name__]
```
Comment (michaelbenayoun) [inline optimum/onnxruntime/modeling_decoder.py:626]: Maybe we can make a method that would do that, and avoid this, which is "too aware" of the underlying implementation IMO.

Something like: 
```
cls._auto_models_to_task(auto_model_class)
```

File: optimum/exporters/onnx/convert.py (hunk 1 above)
Comment (fxmarty) [reply optimum/exporters/onnx/convert.py:62]: Should I just remove the check `issubclass(type(model), classinfo) ` ?

File: optimum/exporters/onnx/convert.py (hunk 2 above)
Comment (fxmarty) [reply optimum/exporters/onnx/convert.py:168]: I use it only here in the end

Comment (fxmarty) [reply optimum/exporters/onnx/convert.py:168]: It is used only twice here, I'd say it's OK

File: setup.py [hunk 4]
```diff
@@ -35,15 +35,13 @@
...
         "protobuf==3.20.1",
-        "diffusers",
     ],
     "onnxruntime-gpu": [
         "onnx",
         "onnxruntime-gpu>=1.9.0",
         "datasets>=1.2.1",
         "evaluate",
         "protobuf==3.20.1",
-        "diffusers",
```
Comment (echarlaix) [inline setup.py:46]: Shouldn't we remove this dependency from `"exporters-tf"` and `"exporters"` as well ?

**Review by echarlaix — COMMENTED** [review]
Thanks for taking care of this @fxmarty !!

File: setup.py (hunk 4 above)
Comment (fxmarty) [reply setup.py:46]: yes

**Review by michaelbenayoun — APPROVED** [review]
LGTM

**Review by JingyaHuang — APPROVED** [review]
LGTM

**Pull request merged**